lib/pdk/validators/metadata/task_metadata_lint.rb

```diff
@@ -63,9 +63,20 @@ def self.schema_file
 
       def self.download_schema_from_forge
         PDK.logger.debug(_('Task Metadata Schema was not found in the cache. Now downloading from the forge.'))
+        ssl_failures ||= 0
         open(FORGE_SCHEMA_URL).read
-      rescue Exception => e
-        raise PDK::CLI::FatalError, _("Unable to download Task Metadata Schema file. Please check internet connectivity and retry this action. %{error}") % { error: e }
+      rescue OpenSSL::SSL::SSLError => e
+        ssl_failures += 1
+
+        unless ssl_failures < 2 && Gem.win_platform? && PDK::Util.gem_install?
```

Comment:
Does this mean that on a Windows system with a package install getting SSL errors, we'll retry forever?

Reply:
See `ssl_retries` variable. Although we might need to do an explicit "begin" context instead of rescuing the whole method.

Reply:
yep, I see the counter. I misread the conditional itself.